### ISSUE TITLE
Add group filter to Custom Reports

### ISSUE THREAD

**Issue by evilforces** [catalyst/moodle-block_dedication#111]
Dear developers, we have been using Course Dedication plugin in several of our training centers for many years and the truth is that we are delighted with its effectiveness. Thank you very much for this great contribution.

We create personalized reports on our students with Moodle's integrated reporting tool, but we realize that if we use the plugin data as a source we cannot filter the students by group.

Would it be possible to add, in addition to the course and student data, the data of the group to which they belong?

Unfortunately, Moodle's custom reports do not allow you to create a report using different sources and if we use the source of this plugin we cannot create filters with the groups.

We thought it would be very useful to have this information in the reports.

Would this be possible?

Thank you very much in advance.

**Comment by danmarsden** [comment]
Great to hear you find the plugin useful! - I agree, the report source should make more entities avaialble and so I've just pushed a fix into the 4.4 branch that adds support for a number of other entities (including groups)

I haven't backported this to the older branches, because it would be more work to do this as the patch required will be slightly different. If you have internal development capabilities and need this for an older moodle release, feel free to submit a PR with the improvements.

**Comment by evilforces** [comment]
Dear Dan, thank you very much for your quick response and above all for your great work adding this functionality to the plugin, which in our opinion is very necessary.

Unfortunately we are in version 4.3 of Moodle and at the moment we cannot migrate to version 4.4 due to the lack of updates to some other plugins necessary for our virtual campus.

We do not have the ability to program the plugin ourselves unfortunately.

Is there no possibility to make these changes in version 4.3?

Thank you very much for your great work.

**Comment by danmarsden** [comment]
there is always a possibility - but it needs someone to spend the time doing the work - unfortunately I've done all I can squeeze in "for free" right now - but at least when you do finally upgrade to 4.4 or higher, you will get the benefit of this improvement.

**Comment by evilforces** [comment]
Thank you very much for your great work and effort Dan.

It is a very useful plugin and we hope it continues its development for a long time.

**Comment by evilforces** [comment]
Dear Dan, I just updated my Moodle installation to version 4.4.1+ to have the latest improvements to your plugin.

I have updated the block_dedication plugin to version 2024052400 but when I run a report it continues without being able to see the group filter options as you mentioned in your previous message.

I am doing something wrong? What can I do to get the latest plugin improvements in my installation?

Thanks for your kindness.
